obsreport/obsreport_exporter.go

```diff
@@ -103,18 +184,56 @@ func (exp *Exporter) startOp(ctx context.Context, operationSuffix string) contex
 	return ctx
 }
 
-func (exp *Exporter) recordMetrics(ctx context.Context, numSent, numFailedToSend int64, sentMeasure, failedToSendMeasure *stats.Int64Measure) {
+func (exp *Exporter) recordMetrics(ctx context.Context, dataType config.DataType, numSent, numFailed int64) {
 	if exp.level == configtelemetry.LevelNone {
 		return
 	}
-	// Ignore the error for now. This should not happen.
-	if numFailedToSend > 0 {
-		_ = stats.RecordWithTags(ctx, exp.mutators, sentMeasure.M(numSent), failedToSendMeasure.M(numFailedToSend))
+	if exp.useOtelForMetrics {
```

Comment:
Any thoughts on removing the need for a `bool` here and instead registering either an otel or oc specific func to call at the initialization of the exporter?

Reply:
In practice may sound better if this is a long term, just for few versions I feel code is easier to delete after the testing phase in the current form.

Reply:
Sounds good, I just didn't want to end up with a bunch of if statements everywhere, but since this is only temporary, i'm ok with this approach